covalent/_workflow/postprocessing.py

```diff
@@ -0,0 +1,170 @@
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+"""Module containing post-processing related functions."""
+
+from builtins import list
+from dataclasses import asdict
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from .._shared_files import logger
+from .._shared_files.context_managers import active_lattice_manager
+from .._shared_files.defaults import (
+    DefaultMetadataValues,
+    postprocess_prefix,
+    prefix_separator,
+    sublattice_prefix,
+)
+from .transport import _TransportGraph, encode_metadata
+
+if TYPE_CHECKING:
+    from .electron import Electron
+
+DEFAULT_METADATA_VALUES = asdict(DefaultMetadataValues())
+
+app_log = logger.app_log
+log_stack_info = logger.log_stack_info
+
+
+class Postprocessor:
+    def __init__(self, lattice) -> None:
+        self.lattice = lattice
+
+    def _is_postprocessable_node(self, tg: _TransportGraph, node_id: int) -> bool:
+        """Filter nodes that should be included in postprocessing."""
+        node_name = tg.get_node_value(node_id, "name")
+        return bool(
+            not node_name.startswith(prefix_separator) or node_name.startswith(sublattice_prefix)
+        )
+
+    def _filter_electrons(self, tg: _TransportGraph, bound_electrons: List) -> List:
+        """Filter bound electrons for ones that should not be included in postprocessing.
+
+        Args:
+            tg: Transport graph of the lattice.
+            bound_electrons: List of bound electrons.
+
+        Returns:
+            List of bound electrons that should be included in postprocessing.
+
+        """
+        filtered_node_ids = [
+            node_id for node_id in tg._graph.nodes if self._is_postprocessable_node(tg, node_id)
+        ]
+        return [bound_electrons[node_id] for node_id in filtered_node_ids]
+
+    def _postprocess(self, *ordered_node_outputs: List[Any]) -> Any:
+        """
+        Post processing function to be called after the lattice execution.
+        This takes care of executing statements that were not an electron
+        but were inside the lattice's function. It also replaces any calls
+        to an electron with the result of that electron execution, hence
+        preventing a local execution of electron's function.
+        Note: Here `node_outputs` is used instead of `electron_outputs`
+        since an electron can be called multiple times with possibly different
+        arguments, but every time it's called, it will be executed as a separate node.
+        Thus, output of every node is used.
```

Comment:
Great explanation!

Reply:
i think it was Casey's :D

Reply:
Actually it was copied from `runner.py` and formerly `execution.py` -- so probably @kessler-frost 's.

Reply:
😂